internal/provider/models/team.go

```diff
@@ -26,7 +26,23 @@ type Team struct {
 	UpdatedBy        types.Object `tfsdk:"updated_by"`
 }
 
-func (data *Team) ReadFromResponse(ctx context.Context, team *iam.Team) diag.Diagnostics {
+type TeamResource struct {
+	Id               types.String `tfsdk:"id"`
+	Name             types.String `tfsdk:"name"`
```

Comment:
maybe we should also add members info to data source
not in this pr but just a suggestion
we can probably add another rest api call: list team members